rows/utils.py

```diff
@@ -297,3 +310,53 @@ def export_to_uri(table, uri, *args, **kwargs):
         raise ValueError('Plugin (export) "{}" not found'.format(plugin_name))
 
     return export_function(table, uri, *args, **kwargs)
+
+
+def decompress(path, **kwargs):
+    """
+    Given a bz2, gzip or lzma file returns a decompressed file object. All
+    kwargs are passed to either `bz2.openn`, `gzip.open` or `lzma.open`.
+    :param path: (str) path to a bz2, gzip or lzma file
+    """
+    filename = os.path.basename(path)
```

Comment:
As the current API accepts filenames or file-objects,
 this function should also do (this decision was inspired in the Python stdlib modules, such as `csv`). You can get some help using [`rows.plugins.utils.get_filename_and_fobj`](https://github.com/turicas/rows/blob/develop/rows/plugins/utils.py#L82).

Reply:
That was indeed pretty helpful, thanks ; )